### ISSUE TITLE
LVM Logical volume not removed or re-used if gnt-instance add failed

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#84]
Originally reported of Google Code with ID 28.

```
<b>What steps will reproduce the problem?</b>

Setup:
 orca is the master node
 dolphin1 is another node
 both don't have the drbd module loaded

1. sudo gnt-instance add --node orca:dolphin1 --disk-template drbd
--os-type debian-etch ns1.example.com
2. Above command will fail with lots of python stack traces including:

  File "/usr/lib/python2.5/site-packages/ganeti/bdev.py", line 984, in
_GetProcData
    stat = open("/proc/drbd", "r")
IOError: [Errno 2] No such file or directory: '/proc/drbd'
failed to create volume sda
(<DRBD8(hosts=orca.example.com-dolphin1.example.com, port=11000, configured
as 192.168.X.X:11000 192.168.X.X:11000,
backend=<LogicalVolume(/dev/orca/c8c1a990-5540-4d61-92b1-1747da39840d.sda_data,
not visible, size=20480m)>,
metadev=<LogicalVolume(/dev/orca/a9ac7951-5a4e-4df5-ac4a-4887b1057d57.sda_meta,
not visible, size=128m)>, visible as /dev/sda, size=20480m)>) on secondary
node dolphin1.example.com!
[...]
could not remove block device sdb on node dolphin1.example.com, continuing
anyway
Failure: command execution error:
Device creation failed, reverting...

3. Check LVM on dolphin:
sudo lvdisplay
[...]
  --- Logical volume ---
  LV Name               
/dev/orca/c8c1a990-5540-4d61-92b1-1747da39840d.sda_data
  VG Name                orca
  LV UUID                0xHf93-7ZOB-n3N7-1UMG-xlcP-R6D6-Ip02GR
  LV Write Access        read/write
  LV Status              available
  # open                 0
  LV Size                20.00 GB
  Current LE             5120
  Segments               1
  Allocation             inherit
  Read ahead sectors     0
  Block device           254:2

  --- Logical volume ---
  LV Name               
/dev/orca/a9ac7951-5a4e-4df5-ac4a-4887b1057d57.sda_meta
  VG Name                orca
  LV UUID                bv1W3f-IYxx-OEPC-moGk-4JnD-GO8i-YVnPsb
  LV Write Access        read/write
  LV Status              available
  # open                 0
  LV Size                128.00 MB
  Current LE             32
  Segments               1
  Allocation             inherit
  Read ahead sectors     0
  Block device           254:3



<b>What is the expected output? What do you see instead?</b>
 LVM Logical volumes should be removed when disk creation doesn't work.
 As it says in the last line: 'Device creation failed, reverting...'

<b>What version of the product are you using? On what operating system?</b>
 Ganeti 1.2.0-1 from ubuntu repositories

<b>Please provide any additional information below.</b>

 
```
Originally added on 2008-05-27 13:34:15 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This has been fixed in the 2.0 release (just checked with 2.0.2 and the LVs are being
removed).

thanks,
iustin
```
Originally added on 2009-08-05 12:07:37 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.0

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:07 +0000 UTC.

Changed State: Released
